Final project: Morgan Kindel and Julia Burke

morgankindel opened the pull request.

jnburke4 committed:
Update README.md

Julia and Morgan's project outline

morgankindel committed:
Update README.md

leej3 commented:
Looks good. I strongly encourage you to use pre-existing tools to achieve all of these details. Your project will be creating a pipeline to run all of those tools. Snakemake or Prefect would be two tools that will help you create such a workflow. 

See my comment [here](https://github.com/biof309/project_spring_2020/pull/7) about working together.

See my comment [here](https://github.com/biof309/project_spring_2020/pull/18) about balancing the project goals of ticking off details on the rubric etc.

leej3 commented:
@jnburke4

jnburke4 committed:
Update README.md

morgankindel committed:
Create ref_images

morgankindel committed:
Add files via upload

morgankindel committed:
Add files via upload

morgankindel committed:
2nd commit

morgankindel committed:
Delete image_processing.py

morgankindel committed:
Delete image_processing.py

morgankindel committed:
Delete Scikit image Setup.ipynb

morgankindel committed:
Delete script.ipynb

morgankindel committed:
Delete .DS_Store

morgankindel committed:
updated reference images

morgankindel committed:
Delete .DS_Store

morgankindel committed:
re-ordered reference images for grid

morgankindel committed:
Delete .DS_Store

morgankindel committed:
updated main script

jnburke4 committed:
New project description 

for julia burke and morgan kindel

morgankindel committed:
update read only file

morgankindel committed:
Merge branch 'master' into master

morgankindel committed:
Merge branch 'master' of https://github.com/morgankindel/project_spring_2020

morgankindel committed:
update sample images

morgankindel committed:
Delete .DS_Store

morgankindel committed:
change sample images 2

jnburke4 committed:
reference images

morgankindel committed:
added tests and seperated main code into functions

morgankindel committed:
Update README.md

morgankindel committed:
Delete 1-checkpoint.jpg

morgankindel committed:
Delete 2-checkpoint.jpg

morgankindel committed:
Delete 3-checkpoint.jpg

morgankindel committed:
Delete img1-checkpoint.jpg

morgankindel committed:
Delete img3-checkpoint.jpg

morgankindel committed:
Delete .DS_Store

morgankindel committed:
Delete .DS_Store

jnburke4 committed:
Add files via upload

jnburke4 committed:
Update README.md

jnburke4 committed:
Delete 1_0.46.jpg

jnburke4 committed:
Delete 9_1.32.jpg

jnburke4 committed:
Delete 8_1.00.jpg

jnburke4 committed:
Delete 7_0.70.jpg

jnburke4 committed:
Delete 6_1.12.jpg

jnburke4 committed:
Delete 2_0.82.jpg

jnburke4 committed:
Delete 5_0.94.jpg

jnburke4 committed:
Delete 3_1.06.jpg

jnburke4 committed:
Delete 4_0.58.jpg

jnburke4 committed:
Create reference

morgankindel committed:
Merge pull request #2 from jnburke4/master

merging final project

morgankindel committed:
Delete coordinate_match-Copy1.ipynb

jnburke4 committed:
Add files via upload

1. Import files

morgankindel committed:
Delete grid.jpg

jnburke4 committed:
Add files via upload

load input image from Sample images file folder or set new path for new input image

jnburke4 committed:
Add files via upload

Create the reference image set up, includes set up to iterate through several images as well as grid for matching template.

jnburke4 committed:
Add files via upload

creates a copy of template grid to put the box on, indicating the most similar image

morgankindel committed:
Merged .py script renamed to burkekindel

jnburke4 committed:
Delete reference

morgankindel committed:
deleted unnessecary files

morgankindel committed:
Merge pull request #3 from jnburke4/master

Merge edits_BurkeKindelproject

morgankindel committed:
Create project_spring_2020_burkekindel

morgankindel committed:
Delete project_spring_2020_burkekindel

leej3 commented:
Can you guys just leave a comment here about how the merging/merge conflicts went. I'd like to get a sense of how much of time you lost to that.

Also your code is nicely modularized in that it is split into notebooks. If you can make functions out off the some of the code there some easy points there. And some docstrings and assertions (or even tests).

morgankindel committed:
Delete 2020pythonspring_julia.ipynb

trying to resolve conflict that won't allow new commits

morgankindel committed:
Merge branch 'master' of https://github.com/morgankindel/project_spring_2020

morgankindel committed:
trying to resolve conflict

morgankindel committed:
Merge branch 'master' of https://github.com/morgankindel/project_spring_2020

morgankindel committed:
Added .new version of main.py, split into functions

morgankindel committed:
moved main_functions.py into _pkg folder

jnburke4 commented:
> Can you guys just leave a comment here about how the merging/merge conflicts went. I'd like to get a sense of how much of time you lost to that.
> 
> Also your code is nicely modularized in that it is split into notebooks. If you can make functions out off the some of the code there some easy points there. And some docstrings and assertions (or even tests).

The merging was surprisingly extremely difficult, I couldn't say how much time we have spent sorting out where to put things but it was many hours. Ultimately, we decided since most of our work was here in this repository, we would post here. We merged some but this was working out better. 
We split functions into tests as well, thanks for the suggestions!

leej3 commented:
Ok good to know thanks. Yea we didn't work through that. There are various strategies to make it easier. It is especially difficult with Jupyter notebooks though.

leej3 commented:
#superceded by #32

leej3 commented:
Excellent job. I appreciate the toils with trying to resolve git merge conflicts. I didn't know, otherwise I could have provided some strategies for improving the situation. With that said you still created a nicely organized, modular project, with lots of comments.

Areas for improvement for your future in coding:

References to pre-existing solutions could be better. You use opencv to solve your problem: mentioning this and the reasons for this decision would be useful for others considering reusing your code. Even if "my lab uses it" that's fine but demonstration of awareness of the context to your decision is important

Look at some of the other students' projects regarding testing: adding tests in the tests directory will make any projects far more scalable and maintainable. Especially in a collaborative setting, tests provide sanity to the development process, ensuring that changes introduced do not break previously working functionality.

Python modules should typically define functions and not run them or much other code (besides imports and some general variable declaration).

If you do wish to run the file as an executable too you can wrap that bit of code like this:

```
__name__ == '__main__':
    filecount()
    create_dict()
    create_template()
    match_images()
```

You have stored a few too many images in the repository. Ideally you would store this somewhere else. It quickly become unmanageable to work with git if you have such a large size on disk. Solutions include:

+ Make the data much smaller (often impractical)
+ storing the images separately on a webserver
+ storing the data in a git submodule
+ using git-annex or datalad

Regarding the "package":
In your setup.py file you are missing a comma after the url argument and you should include your dependencies in the install_requires argument (opencv-python, PIL etc)

Overall your attention to python style is good. A notable exception is lines of code that are too long. Moving forward keep an eye on the pep8 style guide to continue to grow in this respect

Well done.
